Pre-check standard conditions for new offers

steventux opened the pull request:
## Context

Standard conditions are compulsory for a trainee to start a course and are usually included
in an offer. Providers can choose to omit these conditions but they should be included by default.

<!-- Why are you making this change? What might surprise someone about it? -->

## Changes proposed in this pull request

Standard conditions are pre-checked when a provider makes a new offer. The provide can omit them by unchecking the condition(s).

<!-- If there are UI changes, please include Before and After screenshots. -->

### Before

![image](https://user-images.githubusercontent.com/93511/75444528-836bd680-595b-11ea-92cb-78569538a608.png)


### After

![image](https://user-images.githubusercontent.com/93511/75444473-66cf9e80-595b-11ea-9d3a-e975f1757619.png)


## Guidance to review

I considered moving the repetition of checkbox values (they are pretty wordy) to a translation file. The use and guidance of translation files is under discussion though, would be happy to hear reviewer's thoughts on this. It would be DRYer to use translations but given these are form checkbox values I wonder if it's the correct use of i18n?

<!-- How could someone else check this work? Which parts do you want more feedback on? -->

## Link to Trello card

<!-- http://trello.com/123-example-card -->
https://trello.com/c/MpKzZ1HO/1698-make-default-conditions-not-optional-when-making-an-offer-in-the-provider-ui

## Things to check

- [x] This code doesn't rely on migrations in the same Pull Request
- [x] If this code includes a migration adding or changing columns, it also backfills existing records for consistency
- [x] API release notes have been updated if necessary
- [x] New environment variables have been [added to the Azure config](https://github.com/DFE-Digital/apply-for-postgraduate-teacher-training#azure-hosting-devops-pipeline)

mnacos commented on app/controllers/provider_interface/decisions_controller.rb:
```diff
@@ -22,6 +22,7 @@ def new_offer
       @application_offer = MakeAnOffer.new(
         actor: current_provider_user,
         application_choice: @application_choice,
+        standard_conditions: ['Fitness to Teach check', 'Disclosure and Barring Service (DBS) check'],
```
Seems reasonable but may I suggest a small refactoring. How about moving these pre-checked standard conditions to a constant within `MakeAnOffer`, which then becomes the source of truth for what conditions are considered standard, instead of defining this in the controller.

You could then change `MakeAnOffer#initialize` to use that constant as a default value instead of `nil` and drop this line from the controller. There may be some tests that fail as a result (_if_ we have system spec exercising making an offer with no conditions, for example), but it'd be interesting to see what breaks.

steventux replied:
@mnacos good point https://github.com/DFE-Digital/apply-for-postgraduate-teacher-training/pull/1480/commits/e854c7fe6d259dd7d461f2fa40225aa0fbb7ced6 makes those values defaults in the `MakeAnOffer` constructor as advised.

mnacos commented on app/services/make_an_offer.rb:
```diff
@@ -16,7 +16,7 @@ def initialize(
     application_choice:,
     offer_conditions: nil,
-    standard_conditions: nil,
+    standard_conditions: ['Fitness to Teach check', 'Disclosure and Barring Service (DBS) check'],
```
Was hoping for something stubbable, such as
```ruby
  DEFAULT_STANDARD_CONDITIONS = ['Fitness to Teach check', 'Disclosure and Barring Service (DBS) check'].freeze
```
What do you think?

steventux commented:
@mnacos @stevehook thanks for the reviews, I think I've addressed all points now.
I wonder if reusing `MakeAnOffer::STANDARD_CONDITIONS` to populate the values of the checkboxes might make the value definitions less brittle.

mnacos commented:
> @mnacos @stevehook thanks for the reviews, I think I've addressed all points now.
> See [b329c48#diff-541dbd1e1d2e8eee85a7dd9c76c6b235R20](https://github.com/DFE-Digital/apply-for-postgraduate-teacher-training/commit/b329c4858de95625177c12a767c516c7433fada6#diff-541dbd1e1d2e8eee85a7dd9c76c6b235R20)
> and
> [b329c48#diff-002f3970d414b723cc19c694b0d9c902R62](https://github.com/DFE-Digital/apply-for-postgraduate-teacher-training/commit/b329c4858de95625177c12a767c516c7433fada6#diff-002f3970d414b723cc19c694b0d9c902R62)
> 
> I wonder if reusing `MakeAnOffer::STANDARD_CONDITIONS` to populate the values of the checkboxes might make the value definitions less brittle.
> 
> eg
> 
> ```
> <% MakeAnOffer::STANDARD_CONDITIONS.each do |condition| %>                                                                                                  
>   <%= f.govuk_check_box :standard_conditions, condition, label: { text: condition } %>                                                               
> <% end %>
> ```

That's a good idea and consistent with having the service itself be the source of truth for the default standard conditions. I'd probably use `@application_offer.standard_conditions` instead of the constant, but not too bothered.

Please add :)

mnacos commented on app/views/provider_interface/decisions/new_offer.html.erb:
```diff
@@ -19,10 +19,13 @@
+            standard_conditions_checkboxes,
+            :id,
+            :name,
+            legend: { text: 'Standard conditions (optional)', size: 'm', tag: 'h2' },
```
should this say 'optional'?

steventux replied:
Hmm I haven't changed this text from the original copy. They are optional in that there is no validation to ensure these items are selected but I wonder about the wording...

steventux replied:
@mnacos https://github.com/DFE-Digital/apply-for-postgraduate-teacher-training/pull/1480/commits/d1d0ce58bcb7eb33c58ff82c9735268f3b68c24d

mnacos replied:
Thanks, I think it's good to merge.

steventux commented:
@mnacos I forced pushed since your approval, this was to make the test for the check box data less brittle https://github.com/DFE-Digital/apply-for-postgraduate-teacher-training/pull/1480/files#diff-935dc2f0c5c19b13b4a65dc6608f24b0R19-R25
